jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt

```diff
@@ -17,9 +18,15 @@ class ArnPsiReferenceContributor : PsiReferenceContributor() {
             ) {
                 override fun accepts(o: Any?, context: ProcessingContext): Boolean {
                     if (o == null || o !is PsiElement) return false
+                    val manipulator = ElementManipulators.getManipulator(o)
+                    if (manipulator == null) return false
 
-                    if (o.text.contains("arn:")) {
-                        return true
+                    try {
+                        if (manipulator.getRangeInElement(o).substring(o.text).contains("arn:")) {
+                            return true
+                        }
+                    } catch (e: Exception) {
+                        return false
                     }
```

Comment:
Avoidable by checking the element's text length up front?

Reply:
will keep the try/catch, but add explicit element length check